.github/workflows/cli-binary-release.yaml

```diff
@@ -0,0 +1,66 @@
+name: Build and release wgc binaries
+# on:
+#   release:
+#     types: [published]
+on:
+  pull_request:
+    paths:
+      - "cli/**/*"
+      - "connect/**/*"
+      - ".github/workflows/cli-ci.yaml"
+
+permissions:
+  contents: write
+  packages: write
+
+jobs:
+  wgc-releases-matrix:
+    # if: startsWith(github.event.release.tag_name, 'wgc@')
+    name: Build and release wgc binaries
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+
+    strategy:
+      matrix:
+        target: [bun-linux-x64, bun-linux-arm64, bun-darwin-x64, bun-darwin-arm64]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      
+      - uses: oven-sh/setup-bun@v2
+    
+      - uses: ./.github/actions/node
+
+      - name: Install dependencies
+        run: pnpm install --frozen-lockfile
+
+      - name: Build
+        run: pnpm run --filter ./cli --filter ./connect --filter ./shared --filter ./composition build
+
+      - uses: winterjung/split@v2
+        id: split
+        with:
+          separator: "@"
+          msg: "wgc@1.2.3"
+          # msg: "${{ github.event.release.tag_name }}"
+
+      - name: Set BUILD_TIME env
+        run: echo BUILD_TIME=$(date -u +'%Y-%m-%dT%H:%M:%SZ') >> ${GITHUB_ENV}
+
+      - name: "Build wgc binary"
+        working-directory: cli
+        run: bun build --compile --target ${{ matrix.target }} src/index.ts --outfile out/wgc-${{ steps.split.outputs._1 }}-${{ matrix.target }}
+
+      - name: "Upload builds as artifacts"
+        uses: actions/upload-artifact@v4
+        with:
+          name: wgc-${{ steps.split.outputs._1 }}-${{ matrix.target }}
+          path: cli/out/wgc-${{ steps.split.outputs._1 }}-${{ matrix.target }}
+
+      # - name: "Upload build to release"
+      #   working-directory: cli
+      #   env:
+      #     GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      #   shell: bash
+      #   run: gh release upload ${{ github.event.release.tag_name }} wgc-${{ matrix.target }} --clobber
```

Comment:
Please add a test or run the vitest suite with bun so we can catch regressions early